### DISCUSSION TITLE
[Discussion] Make alexa_media a core integration

### DISCUSSION THREAD

**Discussion by phrfpeixoto** [alandtse/alexa_media_player#1048] [Ideas]
Hi,

This is very well known custom integration for HA works GREAT, and it has been features on several video/live streams on major platforms, so why not make this a core integration for Home Assistant?

Has this ever been discussed within the maintainers? Has this ever been discussed with HA's core team?

Having it as part of the core will help get por people using it, grow the platform, and who knows, maybe even get Amazon's attention for documenting local APIs.

**Comment by alandtse** [comment]
We use webscraping for the logon which is prohibited by HA so we'd have to remove the login logic. Also, I don't want to go through the effort of getting HA to approve this integration. It's too big and would require a major rewrite. That said, if someone wants to do the legwork, I'm not opposed.

**Reply by phrfpeixoto** [reply]
I'd love to help, but I've already got two jobs =[
If I find some time, I'll definitely look into it.

Would you have the information on what it takes to get approval, other than the login logic?

**Comment by alandtse** [comment]
If you really want this to happen, you'd likely need to figure out [Amazon oauth](https://gitlab.com/keatontaylor/alexapy/-/issues/13).

**Reply by pove** [reply]
I see on the gitlab thread you linked that you finally managed to do the login into Amazon and keep the token be valid for at least one year. 

Does it mean that the first problem is solved?

**Reply by alandtse** [reply]
In theory yes. We have implemented [oauth](https://gitlab.com/keatontaylor/alexapy/-/issues/13) but are getting login issues in international [jurisdictions](https://github.com/custom-components/alexa_media_player/issues/1101) and particularly if two accounts are [connected](https://github.com/custom-components/alexa_media_player/issues/1098). Also, it's not clear if the reauth issues are happening in general as no one is reporting one way or the other if it stopped asking.

**Comment by mdanilohome** [comment]
Hi @alandtse ! Can you help me? Alexa Media Player integration is crashing frequently. Scripts created with messages that make up an automation, such as an open door warning, unexpectedly stop responding. I restart Home Assistant and in the integration tab, most of the times the message that it has not been loaded appears in the Alexa integration. To get it back up and running, I am required to delete the integration and attempt to install, after entering the two-step verification code on the Amazon website, it remains indefinitely in a "Upload Alexa Media Player Next Step" screen and does not complete an integration. I continue throughout the day, trying again and again until, in a moment, an integration is effected.

**Comment by mdanilohome** [comment]
Hi @alandtse I made the update that was made available today but it didn't work. Alexa Media Player integration stopped working again and ECHO devices are unavailable in Home Assistant. I deleted the integration and try to re-authenticate and the validation on the Amazon website doesn't finish.

**Reply by alandtse** [reply]
Please stop @ing me and hijacking other threads. If you keep doing it I will ban you. I already said what you needed to do to get help. Open a proper issue with actual information per the template.
